sql/core/src/main/scala/org/apache/spark/sql/catalyst/catalog/SQLFunction.scala

```diff
@@ -56,10 +63,48 @@ case class SQLFunction(
   assert((isTableFunc && returnType.isRight) || (!isTableFunc && returnType.isLeft))
 
   override val language: RoutineLanguage = LanguageSQL
+
+  /**
+   * Optionally get the function body as an expression or query using the given parser.
+   */
+  def getExpressionAndQuery(
+      parser: ParserInterface,
+      isTableFunc: Boolean): (Option[Expression], Option[LogicalPlan]) = {
```

Comment:
should it return `Either[Expression, LogicalPlan]`?

Reply:
Either approach works since this is only used in one place. I prefer the current way since it's simpler and more readable :)